libraries/provider_systemd_unit.rb

```diff
@@ -17,6 +17,17 @@ def whyrun_supported?
     action :create do
       unit_path = Systemd::Helpers.unit_path(new_resource)
 
+      if new_resource.drop_in && new_resource.override.nil?
+        fail(
+          Chef::Exceptions::ValidationFailed,
+          'Required argument override is missing!'
+        )
+      end
```

Comment:
well, i was hoping to do something like:

```ruby
    attribute :drop_in, kind_of: [TrueClass, FalseClass], default: false
    attribute :override, kind_of: String, default: nil, required: lazy { |r| r.drop_in }
```

or

```ruby
    attribute :drop_in, kind_of: [TrueClass, FalseClass], default: false
    attribute :override, kind_of: String, default: nil, :callbacks => {
      "is set for drop-in unit" => lambda do |spec|
        !drop_in || !spec.nil?
       end
    }
```

which didn't seem to work... but apparently it did at some time in the past? http://stackoverflow.com/a/24039951

after a whole lot more staring i think i actually finally cracked the code in https://github.com/nathwill/chef-systemd/commit/1075845a413a58698fd3f27eb6f1af3eb8cda2df, at least it seems to work correctly in TK, just weird that *both* methods had to be defined outside the explicitly instead of via the LWRP `attribute` method :/